src/components/socialShare.jsx

```diff
@@ -0,0 +1,55 @@
+import React, {useState, useEffect} from 'react';
+import { ShareIcon } from '@heroicons/react/24/outline';
+
+function SocialShare({ title }) {
+  const [encodedCurrentUrl, setEncodedCurrentUrl] = useState("")
+  const platforms = [
+    {
+      name: "Twitter",
+      baseUrl: "https://twitter.com/share?url",
+      param: "text"
+    },
+    {
+      name: "Facebook",
+      baseUrl: "https://www.facebook.com/sharer/sharer.php?u",
+      param: null
+    },
+    {
+      name: "Linkedin",
+      baseUrl: "https://www.linkedin.com/shareArticle?url",
+      param: "title"
+    },
+  ]
+
+  useEffect(() => {
+    if (typeof window !== "undefined") {
+      // Access the window object only on the client side
+      setEncodedCurrentUrl(encodeURIComponent(window.location.href))
```

Comment:
Question: Does this need to be state? Wouldn't a normal variable suffice here?

Reply:
Using a variable didn't work in the local environment. I didn't investigate further because I found that using global variable like window may have unintended side effects if you have multiple instances of this component on a page. Using state is safer I think.